src/main/java/org/apache/sysds/runtime/instructions/fed/CtableFEDInstruction.java

```diff
@@ -116,119 +120,154 @@ public void processInstruction(ExecutionContext ec) {
 			mo1 = ec.getMatrixObject(input3);
 		}
 
-		long dim1 = Collections.max(Arrays.asList(dims1), Long::compare);
-		boolean fedOutput = dim1 % mo1.getFedMapping().getSize() == 0 && dims1.length == Arrays.stream(dims1).distinct().count();
+		// static non-partitioned output dimension (same for all federated partitions)
+		long staticDim = Collections.max(Arrays.asList(dims1), Long::compare);
+		boolean fedOutput = isFedOutput(mo1.getFedMapping(), mo2);
```

Comment:
We could just replace the boolean _fedOutput_ of the instruction with the __fedOut_ field, right? Then, in principle, the __fedOut_ field is used to decide in which branch (federated/local) the output is created.
However, this change should not affect the _setFedOutput_ method at all, but it would bring with it an advantage for the case, that in the future even partially aggregated federated output is supported. :thinking:

Reply:
I think you are right. The only potential problem is that the current processRequest implementation assumes that mo1 is federated and mo2 is broadcast. This is only a problem if this same logic for checking which input is federated is not a part of the federated planner, which sets the _fedOut field. This is just something I will have to look at later when I extend the federated planner to include this instruction.
The current implementation is sufficient for now.

Reply:
Alright, then I'll just leave the boolean there for now.